traffic_ops/app/lib/API/Tenant.pm

```diff
@@ -0,0 +1,225 @@
+#
+#
+#
+
+use UI::Utils;
+
+use Mojo::Base 'Mojolicious::Controller';
+use Data::Dumper;
+use JSON;
+use MojoPlugins::Response;
+
+my $finfo = __FILE__ . ":";
+
+sub getTenantName {
+	my $self 		= shift;
+	my $tenant_id		= shift;
+	return defined($tenant_id) ? $self->db->resultset('Tenant')->search( { id => $tenant_id } )->get_column('name')->single() : "n/a";
+}
+
+sub isRootTenant {
+	my $self 	= shift;
+	my $tenant_id	= shift;
+	return !defined($self->db->resultset('Tenant')->search( { id => $tenant_id } )->get_column('parent_id')->single());
+}
+
+sub index {
+	my $self 	= shift;
+
+	my @data;
+	my $orderby = $self->param('orderby') || "name";
+	my $rs_data = $self->db->resultset("Tenant")->search( undef, {order_by => 'me.' . $orderby } );
+	while ( my $row = $rs_data->next ) {
+		push(
+			@data, {
+				"id"           => $row->id,
+				"name"         => $row->name,
+				"active"       => $row->active,
+				"parentId"     => $row->parent_id,
+			}
+		);
+	}
+	$self->success( \@data );
+}
+
+
+sub show {
+	my $self = shift;
+	my $id   = $self->param('id');
+
+	my $rs_data = $self->db->resultset("Tenant")->search( { 'me.id' => $id });
+	my @data = ();
+	while ( my $row = $rs_data->next ) {
+		push(
+			@data, {
+				"id"           => $row->id,
+				"name"         => $row->name,
+				"active"       => $row->active,
+				"parentId"     => $row->parent_id,
+			}
+		);
+	}
+	$self->success( \@data );
+}
+
+sub update {
+	my $self   = shift;
+	my $id     = $self->param('id');
+	my $params = $self->req->json;
+
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	my $tenant = $self->db->resultset('Tenant')->find( { id => $id } );
+	if ( !defined($tenant) ) {
+		return $self->not_found();
+	}
+
+	if ( !defined($params) ) {
+		return $self->alert("Parameters must be in JSON format.");
+	}
+
+	if ( !defined( $params->{name} ) ) {
+		return $self->alert("Tenant name is required.");
+	}
+	
+	if ( $params->{name} ne $self->getTenantName($id) ) {
+	        my $name = $params->{name};
+		my $existing = $self->db->resultset('Tenant')->search( { name => $name } )->get_column('name')->single();
+		if ($existing) {
+			return $self->alert("A tenant with name \"$name\" already exists.");
+		}	
+	}	
+
+	if ( !defined( $params->{parentId}) && !$self->isRootTenant($id) ) {
+		return $self->alert("Parent Id is required.");
+	}
+	
+	my $is_active = $params->{active};
+	
+	if ( !$params->{active} && $self->isRootTenant($id)) {
+		return $self->alert("Root user cannot be in-active.");
+	}
+	
+
+	if ( !defined($params->{parentId}) && !isRootTenant($id) ) {
+		return $self->alert("Only the \"root\" tenant can have no parent.");
+	}
+	
+	my $values = {
+		name      => $params->{name},
+		active    => $params->{active},
+		parent_id => $params->{parentId}
+	};
+
+	my $rs = $tenant->update($values);
+	if ($rs) {
+		my $response;
+		$response->{id}          = $rs->id;
+		$response->{name}        = $rs->name;
+		$response->{active}      = $rs->active;
```

Comment:
ensure this returns a true|false in the api. you might have to cast it like \$row->active,